src/pages/api/v1/applications/[id].ts

```diff
@@ -23,65 +21,62 @@ const getApplication = async (req: NextApiRequest, res: NextApiResponse) => {
   }
 };
 
-const updateHandler = async (req: NextApiRequest, res: NextApiResponse) => {
+type UpdateApplicationsRequest = Prisma.ApplicationUpdateInput;
+
+const updateHandler = async (
+  id: number,
+  input: UpdateApplicationsRequest,
+  res: NextApiResponse
+) => {
   try {
-    if (typeof req.query.id === "string") {
-      const { id } = req.query;
-      const { status } = req.body;
-      const data = await prisma.application.update({
-        where: { id: parseInt(id) },
-        data: { status },
-      });
-      if (data) {
-        res.status(200).json(data);
-      }
+    const data = await prisma.application.update({
+      where: { id },
+      data: input,
+    });
+    if (data) {
+      res.status(200).json(data);
     }
   } catch (error) {
-    const { id } = req.query;
-    if (Number(id)) {
-      res.status(404).json({
-        message: `Could not find Application with id ${id}`,
-      });
-    } else {
-      res.status(400).json({
-        message: `Invalid Input: 'id' was not a number`,
-      });
-    }
+    res.status(404).json({
+      message: `Could not find Application with id ${id}`,
+    });
   }
 };
 
-const deleteHandler = async (req: NextApiRequest, res: NextApiResponse) => {
-  const { id } = req.query;
+const deleteHandler = async (id: number, res: NextApiResponse) => {
   try {
-    if (typeof id === "string") {
-      const data = await prisma.application.delete({
-        where: { id: parseInt(id) },
-      });
-      if (data) {
-        res.status(200).json(data);
-      }
+    const data = await prisma.application.delete({
+      where: { id },
+    });
+
+    if (data) {
+      res.status(200).json(data);
     }
   } catch (error) {
-    const { id } = req.query;
-    if (Number(id)) {
-      res.status(404).json({
-        message: `Could not find Application with id ${id}`,
-      });
-    } else {
-      res.status(400).json({
-        message: `Invalid Input: 'id' was not a number`,
-      });
-    }
+    res.status(404).json({
+      message: `Could not find Application with id ${id}`,
+    });
   }
 };
 
 const handler = async (req: NextApiRequest, res: NextApiResponse) => {
+  const { id: rawId } = req.query as { id: string };
+
+  const id = parseInt(rawId);
+
+  if (Number.isNaN(id)) {
+    res.status(400).json({
+      message: `Invalid Input: 'id' was not a number`,
+    });
+  }
```

Comment:
I've moved the logic into the handler function, since it's shared behaviour.

parseInt() will return NaN if the original id is not a number.